Sources/NIOHTTP1/HTTPServerUpgradeHandler.swift

```diff
@@ -180,28 +180,27 @@ public final class HTTPServerUpgradeHandler: ChannelInboundHandler, RemovableCha
         // We'll attempt to upgrade. This may take a while, so while we're waiting more data can come in.
         self.upgradeState = .awaitingUpgrader
 
-        let eventLoop = context.eventLoop
-        let loopBoundContext = context.loopBound
         self.handleUpgrade(context: context, request: request, requestedProtocols: requestedProtocols)
-            .hop(to: eventLoop)  // the user might return a future from another EventLoop.
             .whenSuccess { callback in
-                eventLoop.assertInEventLoop()
                 if let callback = callback {
                     self.gotUpgrader(upgrader: callback)
                 } else {
-                    self.notUpgrading(context: loopBoundContext.value, data: requestPart)
+                    self.notUpgrading(context: context, data: requestPart)
                 }
             }
     }
 
     /// The core of the upgrade handling logic.
     ///
-    /// - Returns: An `EventLoopFuture` that will contain a callback to invoke if upgrade is requested, or nil if upgrade has failed. Never returns a failed future.
+    /// - Returns: An isolated `EventLoopFuture` that will contain a callback to invoke if upgrade is requested,
+    /// or nil if upgrade has failed. Never returns a failed future.
+    /// The `NIOLoopBound` is safe because it's only called after the hop in firstRequestHeadReceived
```

Comment:
What `NIOLoopBound`?

Reply:
This is left over from a previous implementation- removed.